Create pull_request_template.md (#1073)
planetary-social/nos: .github/pull_request_template.md

## Description

> A description of the changes proposed in the pull request. Explain what it does

Reviewer: ```suggestion
A description of the changes proposed in the pull request. Explain what it does to give the reviewer some context for what they're reviewing.
```

> ### How to test

Reviewer: ```suggestion
## How to test
1. Navigate to...
2. Tap...
```

Just want to make sure it's clear what we expect here.

Any testing instruction

> ### Screenshots/Video

Reviewer: ```suggestion
## Screenshots/Video
```

> Skip if this PR contains no visual changes

Reviewer: ```suggestion
Post screenshots or video showing your changes, ideally showing how the app worked before and after these changes. Delete this section if this PR contains no visual changes.
```

> ### Issues covered

Reviewer: ```suggestion
## Issues covered
```

> Add reference(s) to a related issue in your repository.

Reviewer: I'd prefer to have this at the very top.

## Issues covered
Add reference(s) to a related issue in your repository.

## Description
A description of the changes proposed in the pull request. Explain what it does to give the reviewer some context for what they're reviewing.

## How to test
1. Navigate to...
2. Tap...

> Any testing instruction

Reviewer: ```suggestion
```

Reviewer 2: If we are going to write down test steps maybe it makes sense to copy them to the ticket somehow so @setch-l can also use them for UAT?

Reviewer 2: idk I don't necessarily want to create a heavier process or have to manually copy text and then have it duplicated in two places 🤔 maybe I regret making this comment.

## Screenshots/Video

> Post screenshots or video showing your changes, ideally showing how the app worked before and after these changes. Delete this section if this PR contains no visual changes.

Reviewer 2: I don't find this section useful as a reviewer, because I'm going to build and run the app to see for myself. It's also time consuming to create. Are others finding value in this? @martindsq @dcadenas @joshuatbrown

Reviewer: I do use it as a reviewer. I always review the code first, before building and running. When there are screenshots or video, I have a better idea of what I'm reviewing.

As a PR author, I don't find it time consuming to create. I generally test manually a bunch of times; during one of those, I hit the Save Screen or Record Screen option in the Simulator and upload it to the PR.

Reviewer 3: I think this is a good idea as well
